lib/Doctrine/ORM/Cache/DefaultQueryCache.php

```diff
@@ -261,6 +261,14 @@ public function put(QueryCacheKey $key, ResultSetMapping $rsm, $result, array $h
             // @TODO - move to cache hydration components
             foreach ($rsm->relationMap as $alias => $name) {
                 $metadata = $this->em->getClassMetadata($rsm->aliasMap[$rsm->parentAliasMap[$alias]]);
+                $className = $metadata->getName();
```

Comment:
I think the design of the test is broken. The current assertions pass, but further details are indeed skipped